types/index.d.ts

```diff
@@ -0,0 +1,496 @@
+
+    /** Returns a single record from the `collection` if `ids` is a single id, or an array of records if `ids` is an array of ids. */
+    find(id: number | string | number[] | string[]): any;
+
+    /** Returns the first model from `collection` that matches the key-value pairs in the `query` object. */
+    findBy(query: object): any;
+
+    /** Finds the first record matching the provided _query_ in `collection`, or creates a new record using a merge of the `query` and optional `attributesForCreate`. */
+    firstOrCreate(query: object, attributesForCreate?: object): any;
+
+    /** Inserts `data` into the collection. `data` can be a single object or an array of objects. */
+    insert(data: any): any;
+
+    /** Removes one or more records in *collection*. */
+    remove(target?: object | number | string): void;
+
+    /** Updates one or more records in the collection. */
+    update(target: object | number | string, attrs?: object): any;
+
+    /** Returns an array of models from `collection` that match the key-value pairs in the `query` object. */
+    where(query: object): any;
+  }
+}
+
+declare module "miragejs/identity-manager" {
+  /** An IdentityManager is a class that's responsible for generating unique identifiers. You can define a custom identity manager for your entire application, as well as on a per-model basis. */
+  export default class IdentityManager {
+    constructor();
+
+    get(): number;
+
+    /** Registers `uniqueIdentifier` as used. */
+    set(uniqueIdentifier: string | number): void;
+
+    inc(): number;
+
+    /**  Returns the next unique identifier. */
+    fetch(): string;
+
+    /** Resets the identity manager, marking all unique identifiers as available. */
+    reset(): void;
+  }
+}
+
+declare module "miragejs/orm/schema" {
+  import { Collection, ModelInstance } from "miragejs";
+  import Db from "miragejs/db";
+
+  type ModelInitializer<Data> = {
+    [K in keyof Data]: Data[K] extends Collection<infer M>
+      ? Collection<M> | M[]
+      : Data[K];
+  };
+
+  /**
+   * An interface to the Mirage ORM that allows for querying and creating records.
+   */
+  export default class Schema<Registry = Record<string, ModelInstance>> {
+    /** Mirage's in-memory database */
+    readonly db: Db;
+
+    /**
+     * Creates a model of the given type.
+     * @param modelName The type of model to instantiate
+     * @param data Optional initial values for model attributes/relationships
+     */
+    create<
+      K extends keyof Registry,
+      Init extends Registry[K],
+      Data extends Partial<ModelInitializer<Init>>
+    >(
+      modelName: K,
+      data?: Data
+    ): Init & { [K in keyof Init & keyof Data]: Exclude<Init[K], undefined> };
+
+    /** Locates one or more existing models of the given type by ID(s). */
+    find<K extends keyof Registry>(type: K, id: string): Registry[K] | null;
+    find<K extends keyof Registry>(
+      type: K,
+      ids: string[]
+    ): Collection<Registry[K]>;
```

Comment:
Can this ever return `null`? If so, this should be:

```suggestion
    ): Collection<Registry[K]> | null;
```

If it just always returns an empty collection, please ignore this comment! 👍

Reply:
According to the original library code, it can return null.
https://github.com/miragejs/miragejs/blob/master/lib/orm/schema.js#L239-L254
https://github.com/miragejs/miragejs/blob/master/lib/orm/schema.js#L597

Reply:
This is a weird method, in that it can take a single ID or an array of them. If it receives an array, it either returns a Collection containing all the requested items or raises an error, so you'll never get back `null` for this overload.